debezium-core/src/main/java/io/debezium/transforms/ByLogicalTableRouter.java

```diff
@@ -107,78 +86,154 @@
+            .withDescription("The replacement string used in conjunction with " + KEY_FIELD_REGEX.name() +
+                ". This will be used to create the physical table identifier in the record's key.");
+
+    private static final Logger logger = LoggerFactory.getLogger(ByLogicalTableRouter.class);
+    private final AvroValidator schemaNameValidator = AvroValidator.create(logger);
+    private Pattern topicRegex;
+    private String topicReplacement;
+    private Pattern keyFieldRegex;
+    private String keyFieldReplacement;
+    private String keyFieldName;
+    private final Cache<Schema, Schema> keySchemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
+    private final Cache<Schema, Schema> envelopeSchemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
+
+    /**
+     * Logic as follows:
+     * If KEY_FIELD_REGEX has a value that is really a regex, then the KEY_FIELD_REPLACEMENT must be a non-null value.
+     */
+    private static int validateKeyFieldReplacement(Configuration config, Field field, Field.ValidationOutput problems) {
+        String keyFieldRegex = config.getString(KEY_FIELD_REGEX.name());
+        if (keyFieldRegex != null) {
+            keyFieldRegex = keyFieldRegex.trim();
+        }
+
+        String keyFieldReplacement = config.getString(KEY_FIELD_REPLACEMENT.name());
+        if (keyFieldReplacement != null) {
+            keyFieldReplacement = keyFieldReplacement.trim();
+        }
+
+        if (keyFieldRegex == null && keyFieldReplacement == null) {
+            return 0;
+        }
+
+        if (!Objects.equals(keyFieldRegex, "") && keyFieldReplacement == null) {
+            problems.accept(
+                    KEY_FIELD_REPLACEMENT,
+                    null,
+                    String.format("%s must be non-null if %s is set.",
+                            KEY_FIELD_REPLACEMENT.name(),
+                            KEY_FIELD_REGEX.name()
+                    )
+            );
+            return 1;
+        }
+        return 0;
+    }
 
     @Override
     public void configure(Map<String, ?> props) {
         Configuration config = Configuration.from(props);
-        final Field.Set configFields = Field.setOf(TOPIC_REGEX, TOPIC_REPLACEMENT, KEY_FIELD_REGEX,
-                KEY_FIELD_REPLACEMENT);
+        final Field.Set configFields = Field.setOf(
+                TOPIC_REGEX,
+                TOPIC_REPLACEMENT,
+                KEY_FIELD_REGEX,
+                KEY_FIELD_REPLACEMENT
+        );
+
         if (!config.validateAndRecord(configFields, logger::error)) {
             throw new ConnectException("Unable to validate config.");
         }
 
-        topicRegex = Pattern.compile(config.getString(TOPIC_REGEX));
-        topicReplacement = config.getString(TOPIC_REPLACEMENT);
+        topicRegex = Pattern.compile(config.getString(TOPIC_REGEX.name()));
+        topicReplacement = config.getString(TOPIC_REPLACEMENT.name());
```

Comment:
The use of `name()` is not necessary and in fact is less readable. Note that `Configuration` has methods that take a `Field` object (such as [getString(Field)](https://github.com/debezium/debezium/blob/master/debezium-core/src/main/java/io/debezium/config/Configuration.java#L1018)) that call `name()`.